From f45add95a550741bed7aad5a1df62c78163cb079 Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@aries-embedded.de>
Date: Tue, 16 May 2023 16:56:08 +0200
Subject: [PATCH] riscv: dts: m100pfsevp: fix broken emmc support

To permit switching between SD card and eMMC, the clock freqency is
limited to 50 Mhz. HS200 mode will make the eMMC inaccessible.

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 arch/riscv/dts/mpfs-m100pfsevp.dts | 12 ++++--------
 configs/m100pfsevp_defconfig       |  2 ++
 2 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/arch/riscv/dts/mpfs-m100pfsevp.dts b/arch/riscv/dts/mpfs-m100pfsevp.dts
index cfca361eb6..84dbad50ea 100644
--- a/arch/riscv/dts/mpfs-m100pfsevp.dts
+++ b/arch/riscv/dts/mpfs-m100pfsevp.dts
@@ -48,17 +48,13 @@
 
 &mmc {
 	status = "okay";
-	bus-width = <4>;
+	bus-width = <8>;
+	max-frequency = <50000000>;
 	disable-wp;
 	cap-mmc-highspeed;
-	cap-sd-highspeed;
 	card-detect-delay = <200>;
-	mmc-ddr-1_8v;
-	mmc-hs200-1_8v;
-	sd-uhs-sdr12;
-	sd-uhs-sdr25;
-	sd-uhs-sdr50;
-	sd-uhs-sdr104;
+	non-removable;
+	no-1-8-v;
 };
 
 &mac0 {
diff --git a/configs/m100pfsevp_defconfig b/configs/m100pfsevp_defconfig
index 32bf3e8d9d..00357f9869 100644
--- a/configs/m100pfsevp_defconfig
+++ b/configs/m100pfsevp_defconfig
@@ -11,10 +11,12 @@ CONFIG_SBI_V01=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_SYS_LOAD_ADDR=0x80200000
 CONFIG_FIT=y
+CONFIG_BOOTCOMMAND="run mmc_mmc"
 CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_IS_IN_MMC=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
 CONFIG_BOOTP_SEND_HOSTNAME=y
+# CONFIG_MMC_HS200_SUPPORT is not set
 CONFIG_DM_MTD=y
-- 
2.34.1

